plugins/dummy/DummyPortTest.cpp

```diff
@@ -661,7 +661,7 @@ void DummyPortTest::testParamDescription() {
 
   const string description("Code Version");
   size_t str_len = std::min(sizeof(param_description.description),
-                            description.size());
+                            description.size() + 1);
```

Comment:
Hi @yoe so this is intentional, the standard says:

> The Parameter Data Length shall accordingly be set to match the actual size of the text string
> being sent. Parsing of the string shall be length terminated corresponding to the Parameter Data
> Length field. Text fields shall terminate based on Parameter Data Length, however if a NULL is
> encountered then that shall also act as a terminator for the text field.

So 32 characters and no null terminator is valid, hence we don't copy it. What's the GCC error exactly? Anyway it needs overriding locally. I'm surprised and confused it's not triggering the same warning when we're doing exactly the same thing in common/rdm/ResponderHelper.cpp e.g.:
https://github.com/OpenLightingProject/ola/blob/master/common/rdm/ResponderHelper.cpp#L983-L985

Reply:
So I changed this to use GCC pragma's instead, so that the relevant error will be ignored for that line only.

This should fix the issue.